nipype/interfaces/fsl/aroma.py

```diff
@@ -28,7 +28,8 @@ class ICA_AROMAInputSpec(CommandLineInputSpec):
         xor=['feat_dir'],
         desc='volume to be denoised')
     out_dir = Directory(
-        'out', genfile=True, argstr='-o %s', desc='output directory')
+        'out', usedefault=True, genfile=True,
```

Comment:
Judging by `_gen_filename`, this wasn't intended to be a default value. @jdkent can you clarify your intent here?

Reply:
haha, my intent when I made this was to get it to work, I kluged the out_dir definition from [melodic](https://github.com/nipy/nipype/blob/5c3c4137da078a5682e46960c5623b2b4690e692/nipype/interfaces/fsl/model.py#L1606) and [your comment](https://github.com/nipy/nipype/pull/1993/commits/fafdf220669490e57dbf6d2768681b239c74a3be#r114595613) from when I first tried making the wrapper. what a blast from the past!

Reply:
so should `out_dir` have a default value? or should i remove `'out', usedefault=True,`?

Reply:
Okay, so in practice, it's been setting `out_dir=os.getcwd()`, as opposed to `'out'`. Which is the preferable default?

* `out_dir = Directory('out', usedefault=True, mandatory=True, argstr='-o %s', desc='output directory')`
* `out_dir = Directory(genfile=True, argstr='-o %s', desc='output directory')`

I have a slight preference to the former, just to avoid cluttering the local directory (though maybe 'aroma_output' would be a better default). In which case we should also drop `_gen_filename()` method.

Reply:
But it is a default value. It's just specified with a method rather than a trait...

Sorry if I'm being dense, I'm just not following the logic. Anyway, either solution is fine. And @jdkent's is less work.

Reply:
yes, `_gen_filename` will generate an `out_dir`, but I agree with @effigies if we do have a default value it might be better to explicitly do it in the trait definition. Unless you really prefer `os.getcwd` as default.

Reply:
After reading @effigies comment, I agree with him and you. So I reverse my vote, I have a preference for this:
```
out_dir = Directory('out', usedefault=True, mandatory=True, argstr='-o %s', desc='output directory')
```

which means we can delete `_gen_filename`

Reply:
ok, so if we all agree, I'll remove  `_gen_filename` and will use `out` as default

Reply:
Yup, sounds good.

Reply:
@effigies - just an extra question, do we use `mandatory=True` if we already have a default value and `usedefault=True`?

Reply:
I think it's a good idea. The reason I suggest it is because somebody could set it as `Undefined` manually or (god knows why) pipe in a value that fails to be defined from a previous node.

```Python
In [1]: from nipype.interfaces.base import *

In [2]: class A(BaseInterface):
   ...:     class input_spec(TraitedSpec):
   ...:         a = traits.Str('test', usedefault=True)
   ...:     def _run_interface(self, runtime):
   ...:         print(self.inputs.a)
   ...:         return runtime
   ...:     

In [3]: a = A()

In [4]: a.inputs
Out[4]: 

a = test

In [5]: a.inputs.a = Undefined

In [6]: _ = a.run()
<undefined>
```

So unless it really is optional, I would recommend flagging it `mandatory`.

Reply:
sure, I can add it. I was just asking because in many places that we use `usedefault=True`  we don't have `mandatory=True` (and I believe some of them are mandatory). 

But that was just general question, adding  `mandatory=True` won't hurt for sure, so will do it.

Reply:
Ah. Yeah, I think we should probably add `mandatory=True` to those cases, when we see them. Alas, nipype interface metadata will remain inaccurate for a while longer.